src/fs/data/private/spiralratchet.ts

```diff
@@ -11,13 +11,13 @@ interface SpiralRatchet {
 }
 
 
-export async function setup(ctx: EncryptionContext): Promise<SpiralRatchet> {
+export async function setup(ctx: EncryptionContext, seed?: ArrayBuffer, ffSmall?: number, ffMedium?: number): Promise<SpiralRatchet> {
```

Comment:
```suggestion
export async function setup(options: EncryptionContext & { seed?: ArrayBuffer; ffSmall?: number; ffMedium?: number }): Promise<SpiralRatchet> {
```

This is the pattern I go for usually. Originally All the `*Context` types were called `*Options`, but since they were often mandatory at that point, I called them `Context` instead. But `seed`, `ffSmall` and `ffMedium` is exactly what I'd put in the options object :)
Also has the bonus of naming the arguments. So e.g. you wouldn't mix up `ffSmall` and `ffMedium` on the call site.